database/orm/transaction.go

```diff
@@ -0,0 +1,76 @@
+package orm
+
+import "time"
+
+// TransactionType represents transaction type
+type TransactionType uint8
+
+const (
+	Invalid TransactionType = iota + 1
+	BalanceTransfer
+	ValidatorDeposit
+	ValidatorExit
+	AuditorDeposit
+	AuditorExit
+	ObjectCommit
+	ObjectAudit
+	ObjectPor
+)
+
+var (
+	txTypeValue = map[TransactionType]string{
+		Invalid:          "TX_INVALID",
+		BalanceTransfer:  "BALANCE_TRANSFER",
+		ValidatorDeposit: "VALIDATOR_DEPOSIT",
+		ValidatorExit:    "VALIDATOR_EXIT",
+		AuditorDeposit:   "AUDITOR_DEPOSIT",
+		AuditorExit:      "AUDITOR_EXIT",
+		ObjectCommit:     "OBJECT_COMMIT",
+		ObjectAudit:      "OBJECT_AUDIT",
+		ObjectPor:        "OBJECT_POR",
+	}
+
+	txValueType = map[string]TransactionType{
+		"TX_INVALID":        Invalid,
+		"BALANCE_TRANSFER":  BalanceTransfer,
+		"VALIDATOR_DEPOSIT": ValidatorDeposit,
+		"VALIDATOR_EXIT":    ValidatorExit,
+		"AUDITOR_DEPOSIT":   AuditorDeposit,
+		"AUDITOR_EXIT":      AuditorExit,
+		"OBJECT_COMMIT":     ObjectCommit,
+		"OBJECT_AUDIT":      ObjectAudit,
+		"OBJECT_POR":        ObjectPor,
+	}
+)
```

Comment:
use types and names from proto?

Reply:
if the proto changes, the type of all transactions in the database may be  changed

Reply:
Proto cannot change after launch. It is part of consensus and it in State db